js.go

```diff
@@ -2166,6 +2168,19 @@ func (sub *Subscription) Fetch(batch int, opts ...PullOpt) ([]*Msg, error) {
 		defer cancel()
 	}
 
+	deadline, hasDeadline := ctx.Deadline()
```

Comment:
Few lines above, we set `ctx = o.ctx`, which means we use what is set in the option. Then we say `if o.ctx == nil` we create a context with timeout. So I feel like we should, to make it clear, check `if ctx == nil` (instead of o.ctx) and then have a `else` where we check for "hasDeadline" in the user provided context.


The `deadline` can then be gotten down below, before the `ttl = time.Until(deadline)`